samples/apps/copilot-chat-app/webapi/Controllers/ChatParticipantController.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SemanticKernel.Service.Model;
+using SemanticKernel.Service.Storage;
+
+namespace SemanticKernel.Service.Controllers;
+
+/// <summary>
+/// Controller for chat invitations.
+/// This controller is responsible for:
+/// 1. Creating invitation links.
+/// 2. Accepting/rejecting invitation links.
+/// 3. Managing participants in a chat session.
+/// </summary>
+[ApiController]
+[Authorize]
+public class ChatParticipantController : ControllerBase
+{
+    private readonly ILogger<ChatParticipantController> _logger;
+    private readonly ChatParticipantRepository _chatParticipantRepository;
+    private readonly ChatSessionRepository _chatSessionRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ChatParticipantController"/> class.
+    /// </summary>
+    /// <param name="logger">The logger.</param>
+    /// <param name="chatParticipantRepository">The chat participant repository.</param>
+    /// <param name="chatSessionRepository">The chat session repository.</param>
+    public ChatParticipantController(
+        ILogger<ChatParticipantController> logger,
+        ChatParticipantRepository chatParticipantRepository,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._logger = logger;
+        this._chatParticipantRepository = chatParticipantRepository;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Join a use to a chat session given a chat id and a user id.
+    /// </summary>
+    /// <param name="chatParticipantParam">Contains the user id and chat id.</param>
+    [HttpPost]
+    [Route("chatParticipant/join")]
```

Comment:
(nit) At some point, we should think about our api route patterns. For example, I would use "chat/{chatid}/userJoin" as the route. my early thoughts: "{semantic_object}/{id}/{action}".
Please don't be blocked by this comment. (cc @adrianwyatt )

Reply:
Agreed!